src/main.c

```diff
@@ -413,8 +413,12 @@ int ltfs_parse_options(void *priv_data, const char *arg, int key, struct fuse_ar
 		default:
 			if (! priv->first_parsing_pass) {
 				fuse_opt_add_arg(outargs, "-h");
-				if (priv->advanced_help)
-					fuse_main(outargs->argc, outargs->argv, &ltfs_ops, NULL);
+				if (priv->advanced_help){
+					ret = fuse_main(outargs->argc, outargs->argv, &ltfs_ops, NULL);
```

Comment:
@amissael95 

How about this?

Reply:
Hello Abe-San, the following output is from the same change but done on LE code. This output is very similar to the one that will be shown in LTFS/SDE code:

```
# ltfs -a
usage: ltfs mountpoint [options]

general options:
    -o opt,[opt...]        mount options
    -h   --help            print help
    -V   --version         print version

FUSE options:
    -d   -o debug          enable debug output (implies -f)
    -f                     foreground operation
    -s                     disable multi-threaded operation

    -o allow_other         allow access to other users
    -o allow_root          allow access to root
    -o auto_unmount        auto unmount on process termination
    -o nonempty            allow mounts over non-empty file/dir
    -o default_permissions enable permission checking by kernel
    -o fsname=NAME         set filesystem name
    -o subtype=NAME        set filesystem type
    -o large_read          issue large read requests (2.4 only)
    -o max_read=N          set maximum size of read requests

    -o hard_remove         immediate removal (don't hide files)
    -o use_ino             let filesystem set inode numbers
    -o readdir_ino         try to fill in d_ino in readdir
    -o direct_io           use direct I/O
    -o kernel_cache        cache files in kernel
    -o [no]auto_cache      enable caching based on modification times (off)
    -o umask=M             set file permissions (octal)
    -o uid=N               set file owner
    -o gid=N               set file group
    -o entry_timeout=T     cache timeout for names (1.0s)
    -o negative_timeout=T  cache timeout for deleted names (0.0s)
    -o attr_timeout=T      cache timeout for attributes (1.0s)
    -o ac_attr_timeout=T   auto cache timeout for attributes (attr_timeout)
    -o noforget            never forget cached inodes
    -o remember=T          remember cached inodes for T seconds (0s)
    -o nopath              don't supply path if not necessary
    -o intr                allow requests to be interrupted
    -o intr_signal=NUM     signal to send on interrupt (10)
    -o modules=M1[:M2...]  names of modules to push onto filesystem stack

    -o max_write=N         set maximum size of write requests
    -o max_readahead=N     set maximum readahead
    -o max_background=N    set number of maximum background requests
    -o congestion_threshold=N  set kernel's congestion threshold
    -o async_read          perform reads asynchronously (default)
    -o sync_read           perform reads synchronously
    -o atomic_o_trunc      enable atomic open+truncate support
    -o big_writes          enable larger than 4kB writes
    -o no_remote_lock      disable remote file locking
    -o no_remote_flock     disable remote file locking (BSD)
    -o no_remote_posix_lock disable remove file locking (POSIX)
    -o [no_]splice_write   use splice to write to the fuse device
    -o [no_]splice_move    move data while splicing to the fuse device
    -o [no_]splice_read    use splice to read from the fuse device

Module options:

[iconv]
    -o from_code=CHARSET   original encoding of file names (default: UTF-8)
    -o to_code=CHARSET      new encoding of the file names (default: UTF-8)

[subdir]
    -o subdir=DIR           prepend this directory to all paths (mandatory)
    -o [no]rellinks         transform absolute symlinks to relative
32e4b2 LTFS14123W The main function of FUSE returned error (1).

LTFS options:
    -o config_file=<file>     Configuration file (default: /usr/local/etc/ltfs.conf)
    -o work_directory=<dir>   LTFS work directory (default: /tmp/ltfs)
    -o atime                  Update index if only access times have changed
    -o noatime                Do not update index if only access times have changed (default)
    -o tape_backend=<name>    tape backend to use (default: sg)
    -o iosched_backend=<name> I/O scheduler implementation to use (default: unified, use "none" to disable)
    -o dcache_backend=<name>  dentry cache implementation to use (default: ondemand, use "none" to disable)
    -o umask=<mode>           Override default permission mask (3 octal digits, default: 000)
    -o fmask=<mode>           Override file permission mask (3 octal digits, default: 000)
    -o dmask=<mode>           Override directory permission mask (3 octal digits, default: 000)
    -o min_pool_size=<num>    Minimum write cache pool size. Cache objects are 1 MB each (default: 25)
    -o max_pool_size=<num>    Maximum write cache pool size. Cache objects are 1 MB each (default: 50)
    -o rules=<rules>          Rules for choosing files to write to the index partition.
                              The syntax of the rule argument is:
                              size=1M
                              size=1M/name=pattern
                              size=1M/name=pattern1:pattern2:pattern3
                              A file is written to the index partition if it is no larger
                              than the given size AND matches at least one of the name
                              patterns (if specified). The size argument accepts K, M, and G
                              suffixes. Name patterns might contain the special characters
                              '?' (match any single character) and '*' (match zero or more
                              characters)
    -o quiet                  Disable informational messages (same as verbose=1)
    -o trace                  Enable diagnostic output (same as verbose=3)
    -o syslogtrace            Enable diagnostic output to stderr and syslog(same as verbose=303)
    -o fulltrace              Enable full call tracing (same as verbose=4)
    -o verbose=<num>          Override output verbosity directly (default: 2)
    -o eject                  Eject the cartridge after unmount
    -o noeject                Do not eject the cartridge after unmount (default)
    -o sync_type=<type>       Specify sync type (default: time@5)
                              <type> should be specified as follows:
                              time@min:  LTFS attempts to write an index each 'min' minutes.
                                         min should be a decimal number from 1 to 153722867280912930
                                         It is equivalent to "-o sync_type=unmount" when 0 is specified
                                         (default: min=5)
                              close:     LTFS attempts to write an index when a file is closed
                              unmount:   LTFS attempts to write an index when the medium is unmounted
    -o force_mount_no_eod     Skip EOD existence check when mounting (read-only mount)
                              Only use for a CM corrupted medium
    -o device_list            Show available tape devices
    -o rollback_mount=<gen>   Attempt to mount on previous index generation (read-only mount)
    -o strict_label           Require a match between the bar code and the ANSI label when mounting a tape
    -o nostrict_label         Do not require the bar code and the ANSI label to match (default)
    -o release_device         Clear device reservation (should be specified with -o devname or -o changer_devname)
    -a                        Advanced help, including standard FUSE options

LTFS Library options:
    -o library_mode           Run in library mode (not required if changer_devname is specified)
    -o library_cache          Set to use and create caches from the MAM, index and labels (default)
    -o nolibrary_cache        Set to disable the use and creation of MAM, index and labels caches
    -o volser_only_folder     Set to exclude volume name from top level folder name
    -o novolser_only_folder   Set to include volume name in top level folder name (default)
    -o barcode_rules=<rules>  Colon-separated list of barcode patterns. The syntax of each
                              pattern is:
                              +barcode (show matching barcodes)
                              -barcode (blocklist matching barcodes)
                              Patterns may contain the special characters '?' (match any
                              character) and '*' (match zero or more characters).
                              Barcodes are shown or not based on the first matching pattern
                              in the list. If a barcode does not match any pattern, it is
                              shown by default
    -o deactivate_all_drives  Deactivate all tape drives at mount time
    -o activate_drives=D1:D2:D3  Activate specified drives at mount time.
    -o six_char_label_support
                              Allow to load a medium with a six character bar code label
    -o scsi_append_only_mode=<on|off>  Set the tape device append-only mode (default=on)
    -o empty_handed_mode      Deactivate all drives and cartridges at mount time
    -o manual_mount_mode      Enable manual mount mode. Cartridges are not mounted automatically

LTFS Admin Interface options:
    -o admin_port=<num>       TCP/IP port in which to listen for connections (default: 7600)

32e4b2 LTFS17085I Plugin: Loading "lin_tape" tape backend.
IBMTAPE backend options:
    -o devname=<dev>           tape device (default=/dev/IBMtape0)
    -o autodump                enable autodump (default)
    -o noautodump              disable autodump
    -o scsi_lbprotect=<on|off> enable drive logical block protection (default=off)
.
32e4b2 LTFS17085I Plugin: Loading "sg" tape backend.
sg tape backend for IBM tape options:
    -o devname=<dev>           tape device (default=0)
    -o autodump                enable autodump (default)
    -o noautodump              disable autodump
    -o scsi_lbprotect=<on|off> enable drive logical block protection (default=off)
.
32e4b2 LTFS17085I Plugin: Loading "file" tape backend.
FILE backend options:
    -o devname=<dev>          LTFS emulation directory (default=/tmp/ltfs/tape)
    -o emulate_delays         Emulate delays on seeks
    -o file_readonly          Emulate operation in read-only mode
    -o file_p0_warning=<num>  Set early warning position partition 0
    -o file_p1_warning=<num>  Set early warning position partition 1
```

As it shows, when "ltfs -a" is used the new message LTFS14123W specifies that the fuse_main() returned 1, which means "1: Invalid option arguments". 

It means that the correct arguments were not provided to the fuse_main(). I helped me to think that even the use of "fuse_main" is used incorrectly used here.  Do you know what are the right options for the "ltfs" command to provide the fuse_main information correctly? 

It seems like the original use of fuse_main() in line 417 should be changed or even removed, since I have not found any command where the fuse main did not return 1.

Reply:
I'm saying why d you think we need to check the return code of `fuse_main()` here. What's wrong to just ignore it as is?

Reply:
I followed the phytophile to display the return code of `fuse_main()` in every line where it is used, but you are right, I will revert this change, we can simply ignore the error here since the "ltfs -a" command already shows the standard FUSE options correctly.